File: python/perfect-numbers/perfect_numbers.py
def classify(number):
    """ A perfect number equals the sum of its positive divisors.

    :param number: int a positive integer
    :return: str the classification of the input integer
    :raise: ValueError: Classification is only possible for positive integers.
    """
    if number < 1:
        raise ValueError("Classification is only possible for positive integers.")

    divisor_sum = sum([divisor for divisor in range(1, number) if number % divisor == 0])

    if divisor_sum == number:
        return "perfect"
    elif divisor_sum > number:
        return "abundant"
    else:
        return "deficient"

File: javascript/perfect-numbers/perfect-numbers.js
export const classify = (number) => {
  if (number <= 0) {
    throw new Error('Classification is only possible for natural numbers.');
  }

  const aliquotSum = getAliquotSum(number);

  if (aliquotSum === number) {
    return 'perfect';
  } else if (aliquotSum < number) {
    return 'deficient';
  } else {
    return 'abundant';
  }
};

function getAliquotSum(number) {
  let sum = 0;
  for (let i = 1; i <= number / 2; i++) {
    if (number % i === 0) {
      sum += i;
    }
  }
  return sum;
}
